pom.xml

```diff
@@ -753,6 +737,25 @@
           </execution>
         </executions>
       </plugin>
+      <plugin>
+        <groupId>net.revelc.code.formatter</groupId>
+        <artifactId>formatter-maven-plugin</artifactId>
+        <version>${version.formatter.plugin}</version>
+        <configuration>
+          <!-- store outside of target to speed up formatting when mvn clean is used -->
+          <cachedir>.cache/formatter-maven-plugin-${version.formatter.plugin}</cachedir>
+          <configFile>eclipse-format.xml</configFile>
+          <lineEnding>LF</lineEnding>
+          <skip>${format.skip}</skip>
+        </configuration>
+        <dependencies>
+          <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-ide-config</artifactId>
```

Comment:
Overall the formatting changes look good, thanks @xstefank 
I had a look at the rules (https://github.com/quarkusio/quarkus/blob/main/independent-projects/ide-config/src/main/resources/eclipse-format.xml) and I think they look good.
However I don't actually like to be dependent on a quarkus indipendent-project and I would slightly prefer having our formatting rules in our jbosstm project https://github.com/jbosstm/narayana-checkstyle-config/tree/main. That would mean to open a PR there and make a 'narayana-checkstyle-config' release to be able to customize it. Said that I don't think we need to customize those rules at the moment so we can use the quarkus dependency for the moment. WDYT @tomjenkinson , @jmfinelli and @mmusgrov ?

Reply:
That is a great idea, yeah. In my opinion we can decouple it later and not need Martin to change that part now. I do think that we should have the CONTRIBUTING.md reference the expectation (citing a reference to the quarkus one) with this pull request (and we change that later too) so it goes in as close to atomic change as possible but if needed the CONTRIBUTING.md change could be added as a separate commit but we should be ready to go with that.

Reply:
+1